ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcInputFormat.java

```diff
@@ -1727,14 +1727,15 @@ private void populateAndCacheStripeDetails() throws IOException {
     private long computeProjectionSize(List<OrcProto.Type> fileTypes,
         List<OrcProto.ColumnStatistics> stats, boolean[] fileIncluded) throws FileFormatException {
       List<Integer> internalColIds = Lists.newArrayList();
+      int rootColumn = 0;
       if (fileIncluded == null) {
         // Add all.
-        for (int i = 0; i < fileTypes.size(); i++) {
+        for (int i = rootColumn + 1; i < fileTypes.size(); i++) {
           internalColIds.add(i);
         }
       } else {
-        for (int i = 0; i < fileIncluded.length; i++) {
-          if (fileIncluded[i]) {
+        for (int i = rootColumn + 1; i < fileIncluded.length; i++) {
+          if (fileIncluded[i] && (isOriginal || i != OrcRecordUpdater.ROW + 1)) {
```

Comment:
this condition is very vague and hard to understand for an average code reader, please include comments for every parts:
- isOriginal
- OrcRecordUpdater.ROW + 1

Reply:
added. new test